template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ConfigUtils.java

```diff
@@ -63,4 +66,11 @@ public static ApiClusterTemplateVariable variable(String name, String value) {
     public static String getSafetyValveProperty(String key, String value) {
         return String.format(CM_SAFETY_VALVE_PROPERTY_FORMAT, key, value);
     }
+
+    public static Stream<HostgroupView> getHostGroupsWithComponent(String component, TemplatePreparationObject source) {
+        Set<String> namenodeGroups = source.getBlueprintView().getProcessor()
```

Comment:
Not a big deal but `namenodeGroups` is probably not the best name here, as it has no direct relation to NN but rather to `component`.

Reply:
Agreed.  It's leftover after refactoring.

I'm also considering moving this method to `TemplatePreparationObject`, because:

1. both blueprint and hostgroups are taken from that object
2. would be easier to mock

On the other hand, `TemplatePreparationObject` is now only a "holder" + a builder.  What do you think about moving this not too complex logic there?

Reply:
Yeah, makes sense. I do not mind having some small logic in `TemplatePreparationObject`. 😃 BTW the name of that class is also not ideal; I would rather fancy `TemplatePreparationParameters`, `*Context` or something similar. But this is outside the scope of this PR.